aqbanking/pyaqhandler.cpp

```diff
@@ -98,11 +100,42 @@ int PyAqHandler::getPassword(uint32_t flags, const char *token, const char *titl
 }
 
 /**
- * Return always: yes mam!
+ * If no callback always return: yes mam!
  */
 int PyAqHandler::checkCert(const GWEN_SSLCERTDESCR *cd, GWEN_SYNCIO *sio, uint32_t guiid) 
 {
-	// FIXME: implement at least some basic checks => or ask user via callback!
+	// FIXME: implement something better in this case, like GWEN_SslCertDescr_GetStatusFlags
+	if (this->callbackCheckCert == NULL) {
+        fprintf(stderr, "%s", "No certificate check python CB defined! \nThis is insecure!\n");
+        return 0;
+    }
+
+	PyObject *result;
+	const char *commonName = GWEN_SslCertDescr_GetCommonName(cd);
+	const char *statusText = GWEN_SslCertDescr_GetStatusText(cd);
+	const int status = GWEN_SslCertDescr_GetStatusFlags(cd);
+	const char *md5 = GWEN_SslCertDescr_GetFingerPrint(cd);
+
+#if GWENHYWFAR_VERSION_MAJOR > 4 || (GWENHYWFAR_VERSION_MAJOR == 4 && GWENHYWFAR_VERSION_MINOR >= 18)
+	const char *sha512 = GWEN_SslCertDescr_GetFingerPrintSha512(cd);
+
+	PyObject *arglist = Py_BuildValue("({s:s, s:s, s:s, s:s, s:i})", "commonName", commonName, "md5", md5, "sha512", sha512, "statusText", statusText, "status", status);
+#else
+	PyObject *arglist = Py_BuildValue("({s:s, s:s, s:s, s:i})", "commonName", commonName, "md5", md5, "statusText", statusText, "status", status);
+#endif
+	result = PyObject_CallObject(this->callbackCheckCert, arglist);
+	Py_DECREF(arglist);
+
+	if (result == NULL) {
+		fprintf(stderr, "%s", "No answer on cert check returned!\n");
+		PyErr_Print();
+		return GWEN_ERROR_SSL_SECURITY;
+	} else {
+		int cert_valid = PyObject_IsTrue(result);
+		Py_DECREF(result);
+		return cert_valid? 0 : GWEN_ERROR_SSL_SECURITY;
+	}
+
 	return 0;
```

Comment:
```suggestion
```
return not required there anymore - as there is already ensured that a proper return is given (l110, l132, l136).